swap-proxy/src/client_api/mod.rs

```diff
@@ -0,0 +1,228 @@
+                runtime::call_contract(
+                    contract_ref,
+                    (method_names::proxy::METHOD_SET_SWAP_HASH, *swap_hash),
+                )
+            }
+            Self::InsertKYCAllowanceCap(allowance_cap) => {
+                let swap_ref = get_contract_ref();
+                runtime::call_contract(
+                    swap_ref,
+                    (
+                        method_names::proxy::METHOD_INSERT_KYC_ALLOWANCE_CAP,
+                        *allowance_cap,
+                    ),
+                )
+            }
+            Self::InsertSnapshotRecord(ver1_address, amount) => {
+                let swap_ref = get_contract_ref();
+                runtime::call_contract::<_, ()>(
+                    swap_ref.clone(),
+                    (
+                        method_names::proxy::METHOD_INSERT_SNAPSHOT_RECORD,
+                        ver1_address.clone(),
+                        *amount,
+                    ),
+                );
+
+                let contract_purse: URef = runtime::call_contract::<_, URef>(
+                    swap_ref,
+                    (method_names::swap::METHOD_GET_CONTRACT_PURSE,),
+                );
+
+                let transfer_res = system::transfer_from_purse_to_purse(
+                    account::get_main_purse(),
+                    contract_purse,
+                    *amount,
+                );
+
+                match transfer_res {
+                    Ok(_) => (),
+                    Err(err) => runtime::revert(err),
+                }
+            }
+            Self::InsertKYCData(new_mainnet_address, kyc_level) => {
+                let swap_ref = get_contract_ref();
+                runtime::call_contract::<_, ()>(
+                    swap_ref,
+                    (
+                        method_names::proxy::METHOD_INSERT_KYC_DATA,
+                        *new_mainnet_address,
+                        *kyc_level,
+                    ),
+                );
+
+                let transfer_res: TransferResult = system::transfer_to_account(
+                    *new_mainnet_address,
+                    U512::from(BIGSUN_TO_HDAC / 10_u64), // 0.1 Hdac
+                );
+
+                if let Err(err) = transfer_res {
+                    runtime::revert(err);
+                }
+            }
+            Self::UpdateKYCLevel(new_mainnet_address, kyc_level) => {
+                let swap_ref = get_contract_ref();
+                runtime::call_contract(
+                    swap_ref,
+                    (
+                        method_names::proxy::METHOD_UPDATE_KYC_LEVEL,
+                        *new_mainnet_address,
+                        *kyc_level,
+                    ),
+                )
+            }
+            Self::GetToken(swap_contract_hash, ver1_pubkey_arr, message_arr, signature_arr) => {
+                let contract_ref = swap_contract_hash.to_contract_ref().unwrap_or_revert();
```

Comment:
As the comment above, `swap_hash` is already stored in admin's named_key. It seems to be able to load without receiving it from the user.